lib/Dialect/BaseModelica/Transforms/EquationExpressionOpInterfaceImpl.cpp

```diff
@@ -63,24 +505,30 @@ struct TensorFromElementsOpInterface
 
     os << "{";
 
-    llvm::interleaveComma(castedOp.getValues(), os, [&](mlir::Value exp) {
+    llvm::interleaveComma(castedOp.getValues(), os, [&](const mlir::Value exp) {
       ::printExpression(os, exp, inductions);
     });
 
     os << "}";
   }
+
+  bool isEquivalent(mlir::Operation *op, mlir::Operation *other,
+                    mlir::SymbolTableCollection &symbolTableCollection) const {
+    return ::isEquivalent<TensorFromElementsOp>(op, other,
+                                                symbolTableCollection);
+  }
 };
 
 struct TensorBroadcastOpInterface
-    : public EquationExpressionOpInterface::ExternalModel<
-          TensorBroadcastOpInterface, TensorBroadcastOp> {
+    : EquationExpressionOpInterface::ExternalModel<TensorBroadcastOpInterface,
+                                                   TensorBroadcastOp> {
   void printExpression(
       mlir::Operation *op, llvm::raw_ostream &os,
       const llvm::DenseMap<mlir::Value, int64_t> &inductions) const {
     auto castedOp = mlir::cast<TensorBroadcastOp>(op);
 
     os << "{";
-    mlir::TensorType tensorType = castedOp.getResult().getType();
+    const auto tensorType = castedOp.getResult().getType();
```

Comment:
Keep `mlir::TensorType`. There is no cast or template going on here to give hints about the result type.